src/dispatch/plugins/dispatch_google/common.py

```diff
@@ -42,10 +65,15 @@ def get_service(service_name: str, version: str, scopes: list):
 
         delegated_credentials = credentials.with_subject(GOOGLE_SERVICE_ACCOUNT_DELEGATED_ACCOUNT)
 
-        return googleapiclient.discovery.build(
-            service_name,
-            version,
-            credentials=delegated_credentials,
-            cache_discovery=False,
-            developerKey=GOOGLE_DEVELOPER_KEY,
-        )
+        # Prepare arguments to pass to discovery.build() call
+        kwargs = {
+            # Conditionally use delegated credentials when DWD is enabled.
+            "credentials": delegated_credentials if DWD_ENABLED else credentials,
+            "cache_discovery": False,
+        }
+        # Add developerKey only when DWD is enabled
+        if DWD_ENABLED:
```

Comment:
Does the non-DWD path not need a developer key? This was added because without one we would hit the `discovery` (before calling any actual APIs) related rate limit.

Reply:
I will have to test this one more time with the `developerKey`. I never actually hit the rate limit (as I didn't load test it) and I may have had login issues with the `developerKey`. I will test this today and get back to you on this.

Reply:
@kevgliss Since it works with or without the `developerKey`, and adding it adds reliability, I've done away with the (`if DWD_ENABLED`) condition for adding the `developerKey`.